vault/vault.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package vault provides helper functions to improve the go-metrics to stackdriver metric
+// conversions specific to HashiCorp Vault.
+package vault
+
+import "github.com/armon/go-metrics"
+
+// Extractor extracts known patterns from the key into metrics.Label for better metric grouping
+// and to help avoid the limit of 500 custom metric descriptors per project
+// (https://cloud.google.com/monitoring/quotas).
+func Extractor(key []string) ([]string, []metrics.Label) {
+	if len(key) == 2 {
```

Comment:
- should we add validation is len(key) < 1 or > 5?
- nit: seems like this would be better suited as a switch/case statement

Reply:
nit: please explain the '2' with comments

Reply:
I've added a few more comments here